__tests__/order/order.mutation.test.js

```diff
@@ -0,0 +1,347 @@
+/* eslint-disable no-undef */
+const { gql } = require('apollo-boost');
+const client = require('../../utils/apollo-test-client');
+const { 
+  newOrderMutation,
+  newOrderUpdated,
+  orderCreateResult,
+  orderUpdateResult,
+} = require('./order.variables');
+
+let orderId
+
+describe('Order mutations', () => {
+  test('Should create order', async () => {
+    const createOrder = await client.mutate({
+      mutation: gql`
+        mutation($order: OrderInput!) {
+          addOrder(order: $order) {
+            ...on Order {
+              _id
+              status
+              totalPrice {
+                currency
+                value
+              }
+              user {
+                address {
+                  appartment
+                  city
+                  region
+                  street
+                  country
+                  buildingNumber
+                  zipcode
+                }
+                email
+                lastName
+                firstName
+                phoneNumber
+              }
+              items {
+                bottomColor {
+                  lang
+                  value
+                }
+                closure {
+                  lang
+                  value
+                }
+                model {
+                  lang
+                  value
+                }
+                closureColor 
+                size{
+                  widthInCm
+                  weightInKg
+                  heightInCm
+                  volumeInLiters
+                  depthInCm
+                }
+                additions {
+                  lang
+                  value
+                }
+                actualPrice {
+                  currency
+                  value
+                }
+                name {
+                  lang
+                  value
+                }
+                pattern {
+                  lang
+                  value
+                }
+                category {
+                  lang
+                  value
+                }
+                quantity
+                colors {
+                  lang
+                  value
+                }
+                subcategory {
+                  lang
+                  value
+                }
+                bottomMaterial {
+                  lang
+                  value
+                }
+              }
+              delivery {
+                sentBy
+                invoiceNumber
+              }
+              paymentMethod
+            }
+            ...on Error {
+              statusCode
+              message
+            }
+          }
+        }
+      `,
+      variables: { order: newOrderMutation },
+    });
+
+    const order = createOrder.data.addOrder;
+    orderId = order._id;
+    
+    expect(order).toMatchObject(orderCreateResult);
```

Comment:
It is not the best approach to writing tests. I understand that it is much easier to assert the whole object but it makes tests very unreadable. Please, try to assert at least top-level fields separately. For nested values of fields, it is okay to use objects but for root one - no. Thanks

Reply:
fixed